### ISSUE TITLE
Documentation about SSL connection

### ISSUE THREAD

**Issue by edgarjrg** [sequelize/sequelize#11925]
### What was unclear/insufficient/not covered in the documentation

SSL connection is not addressed in the documentation

### If possible: Provide some suggestion on how we can enhance the docs

[Instance Constructor](https://sequelize.org/v5/class/lib/sequelize.js~Sequelize.html#instance-constructor-constructor) could specify all the parameters it actually takes, and if it is just a translation then point to that resource valid options.

### Additional context

[This Issue saved me](https://github.com/sequelize/sequelize/issues/10015#issuecomment-429090656) [as this other one](https://stackoverflow.com/a/44367384/9130688). The last one is a great example of how to add a section to the docs.

### Is this issue dialect-specific?

- [X ] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [X] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by bolmaster2** [comment]
I agree on this one. E.g., the `ssl` property on the sequelize instance options object is missing from the official API reference documentation: https://sequelize.org/master/class/lib/sequelize.js~Sequelize.html#instance-constructor-constructor. I found it by using typescript, from the @types/sequelize (see here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/sequelize/index.d.ts#L5793).

I would say a huge improvement with little work would be to add the `ssl` options property to the documentation.
